alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -2229,14 +2229,29 @@ public static function doAllEntryFlavorsExistInStorage($storageId, $flavorAssets
 	 */
 	public static function getReadyInternalFileSyncsForKey(FileSyncKey $syncKey)
 	{
-		$peridoicStorageFileSyncs = self::getFileSyncsFromPeriodicStorage($syncKey->getPartnerId(), $syncKey);
+		$periodicStorageFileSyncs = self::getFileSyncsFromPeriodicStorage($syncKey->getPartnerId(), $syncKey);
 
 		$c = FileSyncPeer::getCriteriaForFileSyncKey( $syncKey );
 		$c->addAnd ( FileSyncPeer::FILE_TYPE , FileSync::FILE_SYNC_FILE_TYPE_URL, Criteria::NOT_EQUAL);
 		$c->addAnd ( FileSyncPeer::STATUS , FileSync::FILE_SYNC_STATUS_READY );
 		$localfileSyncs = FileSyncPeer::doSelect( $c );
 
-		return array_merge($peridoicStorageFileSyncs, $localfileSyncs);
+		return array_merge($periodicStorageFileSyncs, $localfileSyncs);
+	}
+	
+	public static function getReadyRemoteFileSyncsForAsset($id, $asset, $objectType, $objectSubType)
```

Comment:
Since this function accept also entry as a parameter, this I would change the name $asset to $object, and verify that the input is of type entry or asset.